src/commonMain/kotlin/exchange.dydx.abacus/processor/input/TransferInputProcessor.kt

```diff
@@ -0,0 +1,371 @@
+        exchangeList?.let { data ->
+            exchanges = iMutableListOf()
+            for (i in data.indices) {
+                val item = data[i]
+                val selection = SelectionOption(item.name, item.label, item.label, item.icon)
+                exchanges?.add(selection)
+            }
+        }
+        transfer.withdrawalOptions = WithdrawalInputOptions(
+            needsSize = false,
+            needsAddress = false,
+            needsFastSpeed = false,
+            exchanges = exchanges?.toIList(),
+            chains = chains,
+            assets = assets,
+        )
+    }
+
+    private fun updateTransferOutOptions(
+        transfer: InternalTransferInputState,
+    ) {
+        val chainName = environment?.chainName
+        val chainOption: SelectionOption = if (chainName != null) {
+            SelectionOption(
+                type = "chain",
+                string = chainName,
+                stringKey = null,
+                iconUrl = environment?.chainLogo,
+            )
+        } else {
+            return
+        }
+        val chains: IList<SelectionOption> = iListOf(chainOption)
+
+        val assets: IList<SelectionOption>? = environment?.tokens?.keys?.map { key ->
+            val token = environment.tokens[key]!!
+            SelectionOption(
+                type = key,
+                string = token.name,
+                stringKey = null,
+                iconUrl = token.imageUrl,
+            )
+        }
+        transfer.transferOutOptions = TransferOutInputOptions(
+            needsSize = false,
+            needsAddress = false,
+            chains = chains,
+            assets = assets,
+        )
+    }
+
+    private fun updateTransferToChainType(
+        transfer: InternalTransferInputState,
+        chainType: String
+    ) {
+        val tokenOptions = routerProcessor.tokenOptions(chainType)
+
+        if (transfer.type != TransferType.transferOut) {
+            transfer.tokens = tokenOptions
+            transfer.chain = chainType
+            transfer.token = routerProcessor.defaultTokenAddress(chainType)
+            transfer.chainResources = routerProcessor.chainResources(chainType)
+            transfer.tokenResources = routerProcessor.tokenResources(chainType)
+        }
+        transfer.exchange = null
+        transfer.size = TransferInputSize.safeCreate(transfer.size).copy(size = null)
+        transfer.route = null
+        // needed to pass tests, remove later
+        transfer.depositOptions = DepositInputOptions.safeCreate(transfer.depositOptions)
+            .copy(assets = tokenOptions.toIList())
+        transfer.withdrawalOptions = WithdrawalInputOptions.safeCreate(transfer.withdrawalOptions)
+            .copy(assets = tokenOptions.toIList())
+    }
+
+    private fun updateTransferToTokenType(
+        transfer: InternalTransferInputState,
+        tokenAddress: String
+    ) {
+        // Code not used
```

Comment:
Can you expand on when this code is not being used? 🤔

Reply:
Line 333 to 355 is already called before this function is called.

`resources.tokenSymbol `and `resources.tokenDecimals` are not read, so there is no need to set them.